streamer/src/nonblocking/quic.rs

```diff
@@ -341,14 +341,24 @@ async fn run_server(
                 .total_incoming_connection_attempts
                 .fetch_add(1, Ordering::Relaxed);
 
-            // first do per IpAddr rate limiting
-            if rate_limiter.len() > CONNECTION_RATE_LIMITER_CLEANUP_SIZE_THRESHOLD {
-                rate_limiter.retain_recent();
+            // check overall connection request rate limiter
+            if overall_connection_rate_limiter.current_tokens() == 0 {
+                debug!(
```

Comment:
Please update the metrics on connection throttling.  connection_rate_limited_across_all and connection_rate_limited_per_ipaddr